packages/medusa/src/services/payment-provider.ts

```diff
@@ -281,20 +278,23 @@ export default class PaymentProviderService extends TransactionBaseService {
 
       const context = this.buildPaymentContext(sessionInput)
 
-      const sessionData = await provider
+      const paymentResponse = await provider
         .withTransaction(transactionManager)
         .updatePayment(paymentSession.data, context)
 
-      const amount = this.featureFlagRouter_.isFeatureEnabled(
-        OrderEditingFeatureFlag.key
+      const sessionData = paymentResponse.session_data ?? paymentResponse
+
+      await this.processUpdateRequestsData(
```

Comment:
question: Any chance we could avoid this from firing every time a session is updated?

Reply:
It does nothing if there is not customer data. It is included in the function